lib/crypto.js

```diff
@@ -13,7 +13,41 @@ const ec_P256 = new EC("p256")
 const ec_secp256k1 = new EC("secp256k1")
 
 module.exports = {
-
+  /**
+   * Create a hash digest from the data with an hash algorithm identification prepending the digest
+   * @param {String} seed Keypair derivation seed
+   * @param {Integer} index Number to identify the order of keys to generate
+   * @param {String} curve Elliptic Curves(ed25519, P256, secp256k1)
+   * @param {String} algo Hash algorithm ("sha256", "sha512", "sha3-256", "sha3-512", "blake2b")
+   */
+  deriveAddress: function(seed, index, curve = "ed25519", hashAlgo = "sha256") {
+    const { publicKey } = this.deriveKeyPair(seed, index, curve)
+    switch(curve) {
+      case "ed25519":
+          return concatUint8Arrays(
+              [
+                  Uint8Array.from([0]),
+                  Uint8Array.from(this.hash(publicKey,hashAlgo))
```

Comment:
To avoid duplication of `this.hash(publicKey.hashAlgo)` , you could use the `switch` to return the curve_id and then compute the address outside, so it will be more a DRY approach

Reply:
changed address derivation function

`const { publicKey } = this.deriveKeyPair(seed, index, curve)
    
    switch(curve) {
      case "ed25519":
        curve_id = 0;
        break;
          
      case "P256":
        curve_id = 1;
        break;

      case "secp256k1":
        curve_id = 2;
        break;

      default :
          throw "curve not supported "
    }

    hashedPublicKey = this.hash(publicKey,hashAlgo)
    
    return concatUint8Arrays(
            [
                Uint8Array.from([curve_id]),
                Uint8Array.from(hashedPublicKey)
            ]
        )`